### ISSUE TITLE
Failed to start new created app-operator

### ISSUE THREAD

**Issue by feli5** [operator-framework/operator-sdk#1494]
## Bug Report

**What did you do?**
I created a app-operator refer to the Readme. And try to start up the operator.

**What did you expect to see?**
App operator start normally

**What did you see instead? Under which circumstances?**
```
# kubectl get all
NAME                                READY   STATUS   RESTARTS   AGE
pod/app-operator-75d9c94849-6mqns   0/1     Error    3          52s

NAME                 TYPE        CLUSTER-IP   EXTERNAL-IP   PORT(S)   AGE
service/kubernetes   ClusterIP   10.96.0.1    <none>        443/TCP   61d

NAME                           DESIRED   CURRENT   UP-TO-DATE   AVAILABLE   AGE
deployment.apps/app-operator   1         1         1            0           55s

NAME                                      DESIRED   CURRENT   READY   AGE
replicaset.apps/app-operator-75d9c94849   1         1         0       55s

# kubectl logs pod/app-operator-75d9c94849-6mqns
standard_init_linux.go:190: exec user process caused "permission denied"
```
BTW, my docker image can be found from feli5/app-example

**Environment**
* operator-sdk version: 
operator-sdk version: v0.8.0, commit: 78c472461e75e6c64589cfadf577a2004b8a26b3

* go version: 
go version go1.12.5 linux/amd64

* Kubernetes version information:
Client Version: version.Info{Major:"1", Minor:"14", GitVersion:"v1.14.2", GitCommit:"66049e3b21efe110454d67df4fa62b08ea79a19b", GitTreeState:"clean", BuildDate:"2019-05-16T16:23:09Z", GoVersion:"go1.12.5", Compiler:"gc", Platform:"linux/amd64"}
Server Version: version.Info{Major:"1", Minor:"12", GitVersion:"v1.12.5", GitCommit:"51dd616cdd25d6ee22c83a858773b607328a18ec", GitTreeState:"clean", BuildDate:"2019-01-16T18:14:49Z", GoVersion:"go1.10.7", Compiler:"gc", Platform:"linux/amd64"}

* Kubernetes cluster kind: 

* Are you writing your operator in ansible, helm, or go?
kube-spray

**Possible Solution**

**Additional context**

**Comment by camilamacedo86** [comment]
Hi @feli5,

I understand that you are following the steps here: https://github.com/operator-framework/operator-sdk#create-and-deploy-an-app-operator

I am not able to face the same issue and all works for me. I used Minishift and the system admin user.

**Comment by feli5** [comment]
@camilamacedo86,
Thanks for your reply.
Can you share me your built docker image to let me have a test?

**Comment by joelanford** [comment]
@feli5 This sounds similar to https://github.com/operator-framework/operator-sdk/issues/1206. Is it possible you have a restrictive umask set when you run `operator-sdk build`?

**Comment by feli5** [comment]
After some dig, it seems because there is no user with uid 1001
```
# docker exec -it 5949f84398fb bash
bash-4.2$ whoami
whoami: cannot find name for user ID 1001
```

And if I change to root, it works.
```
# docker exec -it -u root 5949f84398fb bash
bash-4.2# mkdir /var/run/secrets/kubernetes.io/serviceaccount
```
Need to copy secrets to /var/run/secrets/kubernetes.io/serviceaccount
```
bash-4.2# export WATCH_NAMESPACE=default
bash-4.2# export POD_NAME=app-operator-75d9c94849-6mqns
bash-4.2# export OPERATOR_NAME=app-operator
bash-4.2# export KUBERNETES_PORT_443_TCP_PROTO=tcp
bash-4.2# export KUBERNETES_PORT_443_TCP_PORT=443
bash-4.2# export KUBERNETES_PORT_443_TCP_ADDR=10.96.0.1
bash-4.2# export KUBERNETES_SERVICE_HOST=10.96.0.1
bash-4.2# export KUBERNETES_SERVICE_PORT=443
bash-4.2# export KUBERNETES_SERVICE_PORT_HTTPS=443
bash-4.2# export KUBERNETES_PORT=tcp://10.96.0.1:443
bash-4.2# export KUBERNETES_PORT_443_TCP=tcp://10.96.0.1:443
bash-4.2# export OPERATOR=/usr/local/bin/app-operator
bash-4.2# export container=oci
bash-4.2# export PATH=/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin^C
bash-4.2# export USER_UID=1001
bash-4.2# export USER_NAME=app-operator
bash-4.2# app-operator
{"level":"info","ts":1559399403.634943,"logger":"cmd","msg":"Go Version: go1.12.5"}
{"level":"info","ts":1559399403.6349921,"logger":"cmd","msg":"Go OS/Arch: linux/amd64"}
{"level":"info","ts":1559399403.6349971,"logger":"cmd","msg":"Version of operator-sdk: v0.8.0"}
{"level":"info","ts":1559399403.6351745,"logger":"leader","msg":"Trying to become the leader."}
{"level":"info","ts":1559399403.7024112,"logger":"leader","msg":"No pre-existing lock was found."}
{"level":"info","ts":1559399403.713179,"logger":"leader","msg":"Became the leader."}
{"level":"info","ts":1559399403.7495632,"logger":"cmd","msg":"Registering Components."}
{"level":"info","ts":1559399403.7497835,"logger":"kubebuilder.controller","msg":"Starting EventSource","controller":"appservice-controller","source":"kind source: /, Kind="}
{"level":"info","ts":1559399403.749956,"logger":"kubebuilder.controller","msg":"Starting EventSource","controller":"appservice-controller","source":"kind source: /, Kind="}
{"level":"info","ts":1559399403.8155308,"logger":"metrics","msg":"Metrics Service object created","Service.Name":"app-operator","Service.Namespace":"default"}
{"level":"info","ts":1559399403.8155599,"logger":"cmd","msg":"Starting the Cmd."}
{"level":"info","ts":1559399403.915885,"logger":"kubebuilder.controller","msg":"Starting Controller","controller":"appservice-controller"}
{"level":"info","ts":1559399404.0160387,"logger":"kubebuilder.controller","msg":"Starting workers","controller":"appservice-controller","worker count":1}
```

**Comment by feli5** [comment]
BTW, it works, if I comment out the latest line in Dockerfile.
```
FROM registry.access.redhat.com/ubi7/ubi-minimal:latest

ENV OPERATOR=/usr/local/bin/app-operator \
    USER_UID=1001 \
    USER_NAME=app-operator

# install operator binary
COPY build/_output/bin/app-operator ${OPERATOR}

COPY build/bin /usr/local/bin
RUN  /usr/local/bin/user_setup

ENTRYPOINT ["/usr/local/bin/entrypoint"]

#USER ${USER_UID}
```

**Comment by joelanford** [comment]
@feli5 Were you ever able to resolve this without commenting out the `USER` line in your Dockerfile?

We recommend not using the root user for security reasons. The purpose of the`/usr/local/bin/user_setup` script in the Dockerfile is to create a user with UID = 1001.

Looking at the your image, I see a few things that hint that the problem is permissions related:
1. It doesn't look like the user_setup script ran successfully when the Dockerfile ran it during `operator-sdk build`.
2. The permissions of the files in `/usr/local/bin/` are `0700`, which tells me that the permissions of those files on your filesystem are likely too restrictive for what the SDK expects (see #1206). They likely need to be group readable and executable (i.e. 0750).

As I mentioned before, my guess would be that your umask was set to `0077` when you created your project. If you change it to `0022` or `0027` and recreate your project, I would expect that to resolve your issue.

**Comment by joelanford** [comment]
I'm pretty sure that's the issue, but if not, ping me and we can re-open this issue.

**Comment by feli5** [comment]
I changed umask to 0022, but still meet this problem.
```
# umask
0022
# operator-sdk build feli5/app-operator
INFO[0000] Building OCI image feli5/app-operator
Sending build context to Docker daemon  140.7MB
Step 1/7 : FROM registry.access.redhat.com/ubi7/ubi-minimal:latest
 ---> 8d0998e077d3
Step 2/7 : ENV OPERATOR=/usr/local/bin/app-operator     USER_UID=1001     USER_NAME=app-operator
 ---> Running in dca29af90ac7
Removing intermediate container dca29af90ac7
 ---> 98bc9adb7771
Step 3/7 : COPY build/_output/bin/app-operator ${OPERATOR}
 ---> eb580213c180
Step 4/7 : COPY build/bin /usr/local/bin
 ---> 5be2da211cf0
Step 5/7 : RUN  /usr/local/bin/user_setup
 ---> Running in 7c7d12c61868
+ mkdir -p /root
+ chown 1001:0 /root
+ chmod ug+rwx /root
+ chmod g+rw /etc/passwd
+ rm /usr/local/bin/user_setup
Removing intermediate container 7c7d12c61868
 ---> 8f45a8e49166
Step 6/7 : ENTRYPOINT ["/usr/local/bin/entrypoint"]
 ---> Running in 08162042734f
Removing intermediate container 08162042734f
 ---> 940a2bec11a7
Step 7/7 : USER ${USER_UID}
 ---> Running in e0d771bca177
Removing intermediate container e0d771bca177
 ---> f683c9e517cf
Successfully built f683c9e517cf
Successfully tagged feli5/app-operator:latest
INFO[0003] Operator build complete.
```
When I start this new build image, below is what I can find, app-operator's permission is still not correct.
```
# docker run -it --rm -u root --entrypoint /bin/bash feli5/app-operator
bash-4.2# ls /usr/local/bin/
app-operator  entrypoint
bash-4.2# ls -l /usr/local/bin/
total 46924
-rwx------. 1 root root 48042780 Jun 26 01:47 app-operator
-rwx------. 1 root root      373 May 28 16:26 entrypoint
```
